test/NuGet.Core.Tests/NuGet.Synchronization.Test/SynchronizationTest.cs

```diff
@@ -0,0 +1,494 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Test.Utility;
+using SynchronizationTestApp;
+using Xunit;
+
+namespace NuGet.Commands.Test
+{
+    public class SynchronizationTests
+    {
+        private int _value1 = 0;
+        private int _value2 = 0;
+        private SemaphoreSlim _waitForEverStarted = new SemaphoreSlim(0, 1);
+
+        private readonly int DefaultTimeOut = (int)TimeSpan.FromMinutes(5).TotalMilliseconds;
+
+        public SynchronizationTests()
+        {
+            // Delete lock files also on non-Windows platforms.
+            Environment.SetEnvironmentVariable("NUGET_ConcurrencyUtils_DeleteOnClose", "1");
```

Comment:
NuGet 6.x is targeting .NET SDK 6.0, and since the .NET runtime only fixed their bug in .NET 7, it's not safe for us to set this for all tests in the same process until NuGet's dev branch starts targeting .NET 7 (and actually using that TFM to execute tests with).